src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -138,18 +141,22 @@ def __ensure_message_restriction_compliance(self, full_message):
 
     def write_event(self, message, event_level=Constants.TelemetryEventLevel.Informational, task_name=Constants.TELEMETRY_TASK_NAME):
         """ Creates and writes event to event file after validating none of the telemetry size restrictions are breached """
-        if self.events_folder_path is None or not os.path.exists(self.events_folder_path) or not Constants.TELEMETRY_ENABLED_AT_EXTENSION:
+        if self.events_folder_path is None or not os.path.exists(self.events_folder_path) or not Constants.TELEMETRY_ENABLED_AT_EXTENSION or not self.__is_telemetry_startup:
```

Comment:
why is all of this being checked with every single write_event call? The first two parameters should be checked once at init and mapped to the agent_supports_telemetry() / equivalent function talked about on the call.

Reply:
This PR is on top of the telemetry core, which is why these changes appear here